datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java

```diff
@@ -175,8 +191,19 @@ public KafkaMirrorMakerConnectorTask(KafkaBasedConnectorConfig config, Datastrea
 
   @Override
   protected void consumerSubscribe() {
-    LOG.info("About to subscribe to source: {}", _mirrorMakerSource.getTopicName());
-    _consumer.subscribe(Pattern.compile(_mirrorMakerSource.getTopicName()), this);
+    if (_enablePartitionAssignment) {
+      Set<TopicPartition> topicPartition  = getAssignedTopicPartitionFromTask();
```

Comment:
There is a possibility that the topicPartitions assigned to this task will be empty during the first assignment. How is that handled?

Reply:
Its fine for Kafka to subscribe for an emptyList(), so this will work.

But the Kafka cannot perform poll if partition list is empty(), and that's why there is another check around poll. 

I will add additional document on this behavior.